### ISSUE TITLE
pyright should also give error for the type alias set multiple `ParamSpec`-based type parameters not just for the type alias set multiple `TypeVarTuple`-based type parameters

### ISSUE THREAD

**Issue by hyperkai** [microsoft/pyright#11214]
*Memo:
- pyright test.py
- pyright 1.1.407
- Python 3.14.0
- Windows 11

Defining the type alias set multiple [ParamSpec](https://docs.python.org/3/library/typing.html#typing.ParamSpec)-based type parameters, pyright doesn't give error while defining the type alias set multiple [TypeVarTuple](https://docs.python.org/3/library/typing.html#typing.TypeVarTuple)-based type parameters, pyright gives the error as shown below so pyright should also give error for the type alias set multiple `ParamSpec`-based type parameters:

### Multiple `ParamSpec` based:

```python
''' New syntax '''
type TA[**P1, **P2] = Callable[P1, None] | Callable[P2, None] # No error
''' New syntax '''

''' Old syntax '''
# from typing import ParamSpec, TypeAlias

# P1 = ParamSpec('P1')
# P2 = ParamSpec('P2')

# TA: TypeAlias = Callable[P1, None] | Callable[P2, None] # No error
''' Old syntax '''
```

### Multiple `TypeVarTuple` based:

```python
''' New syntax '''
type TA[*Ts1, *Ts2] = tuple[*Ts1] | tuple[*Ts2] # Error
''' New syntax '''

''' Old syntax '''
# from typing import TypeAlias, TypeVarTuple

# Ts1 = TypeVarTuple('Ts1')
# Ts2 = TypeVarTuple('Ts2')

# TA: TypeAlias = tuple[*Ts1] | tuple[*Ts2] # Error
''' Old syntax '''
```

> error: Type alias can have at most one TypeVarTuple type parameter but received multiple ("Ts1", "Ts2")

**Comment by erictraut** [comment]
Pyright is working correctly here and in compliance with the Python typing spec. A type alias (or any generic type) can be parameterized with multiple ParamSpecs, but only one TypeVarTuple is allowed. Refer to the typing spec for details.
